discovery/kubernetes_api_discovery.go

```diff
@@ -48,37 +48,48 @@ func NewK8sAPIDiscoverer(kubeHost string, kubePort int, namespace string, timeou
 func (k *K8sAPIDiscoverer) servicesForNode(hostname, ip string) []service.Service {
 	var services []service.Service
 
-	for _, item := range k.discoveredSvcs.Items {
-		// We require an annotation called 'ServiceName' to make sure this is
-		// a service we want to announce.
-		if item.Metadata.Labels.ServiceName == "" {
-			continue
-		}
+	for svcName, podList := range k.discoveredPods {
+		for _, pod := range podList {
```

Comment:
Since `pod` is a `*K8sPod`, I wonder if it can ever be nil. Guess not, but the k8s API is really weird in some cases.